api/v1beta1/neutronapi_types.go

```diff
@@ -194,6 +198,14 @@ type NeutronAPIDebug struct {
 	Service bool `json:"service"`
 }
 
+// NeutronRoute is used to define all the information for the OpenShift route
```

Comment:
This kind of looks duplicating with Names(i.e Route within Route), Route.RouteAnnotations, if it's done to have more similar annotations in future for different resources, couldn't it be better to have like Annotations.Route, Annotations.Service etc? Or Just have all in top level CRD like RouteAnnotations map[string]string, ServiceAnnotations map[string]string?
Not sure if it's already pointed out and i just missed that discussion.

Also please add functional test for this.

Reply:
Wrt  "NeutronRoute is used to define all the information for the OpenShift route", what all information you mean here? it means just annotations or some other option too?